src/main/java/com/iexec/common/chain/eip712/EIP712Entity.java

```diff
@@ -84,4 +87,24 @@ public List<TypeParam> getDomainTypeParams() {
         return new ArrayList<>(types.get(EIP712Domain.primaryType));
     }
 
+    /**
+     * Builds an authorization token from given {@link Credentials}.
+     * <p>
+     * An authorization token is the concatenation of the following values,
+     * delimited by an underscore:
+     * <ol>
+     *     <li>Entity hash;</li>
+     *     <li>Signed message;</li>
+     *     <li>Credentials address.</li>
+     * </ol>
+     *
+     * @param credentials Credentials representing the signer.
+     * @return The authorization token.
+     */
+    public String buildAuthorizationTokenFromCredentials(Credentials credentials) {
```

Comment:
Would you be able to provide a successful unit test for it?

Reply:
Sure, see https://github.com/iExecBlockchainComputing/iexec-common/pull/335/commits/590204b690bea9fbd91f594250d5b3a33f5c8b0e.